disorder-core/src/Disorder/Core/Gen.hs

```diff
@@ -188,3 +193,12 @@ vectorOfUnique = vectorOfUnique' 30
 -- 'vectorOfUnique'' for details.
 listOf1Unique :: Eq a => Gen a -> Gen [a]
 listOf1Unique g = sized $ \s -> choose (1, max 1 s) >>= (flip vectorOfUnique g)
+
+
+genNonEmpty :: Arbitrary a => Gen (NonEmpty a)
+genNonEmpty =
+  (:|) <$> arbitrary <*> arbitrary
```

Comment:
I feel like this would be more flexible if it took a `Gen a` rather than using an `Arbitrary` constraint. Up to you though.